File: src/main/java/com/liquor/sort/BubbleSort.java
package com.liquor.sort;

import java.util.Arrays;

public class BubbleSort {

    public static void main(String[] args) {
        int[] arr = new int[]{2, 4, 3, 6, 7, 5, 2, 1, 9, 10};

        final int length = arr.length;

        int temp;
        for (int i = 0; i < length - 1; i++) {
            for (int j = i; j < length - 1; j++) {

                if (arr[i] > arr[j]) {
                    temp = arr[i];
                    arr[i] = arr[j];
                    arr[j] = temp;
                }
            }
        }

        System.out.println(Arrays.toString(arr));
    }
}

File: src/main/scala/com/loquor/sort/BubbleSort.scala
package com.loquor.sort

object BubbleSort {

  def main(args: Array[String]): Unit = {
    val arr = Array(2, 4, 3, 6, 7, 5, 2, 1, 9, 10)
    val length = arr.length

    def swap(a: Int, b: Int): Unit = {
      val temp = arr(a)
      arr(a) = arr(b)
      arr(b) = temp
    }

    for (i <- 0 until length) {
      for (j <- i until length) {
        if (arr(i) > arr(j)) {
          swap(i, j)
        }
      }
    }

    println(arr.mkString("Array(", ", ", ")"))
  }
}
